src/cmd_sym.ml

```diff
@@ -41,54 +46,43 @@ let assert_extern_module : Symbolic.P.extern_func Link.extern_module =
 let names = [| "plop"; "foo"; "bar" |]
 
 let symbolic_extern_module : Symbolic.P.extern_func Link.extern_module =
-  let counter = ref 0 in
+  let sprintf = Printf.sprintf in
+  let next_sym_id () : int Choice.t =
+    let/ cnt = Choice.with_thread Thread.sym_cnt in
```

Comment:
Actually, I believe there's no need to put `sym_cnt` in `Thread`. Using a global `int Atomic.t` should be enough, right ?